src/lib/map.c

```diff
@@ -1,2 +1,59 @@
 #include "lib/map.h"
 
+size_t fnv_1a(char *ptr, size_t size) {
+    size_t hash = 0x811C9DC5;
+
+    for (;size--;) {
+        hash ^= ptr[size];
+        hash *= 0x01000193;
+    }
+
+    return hash;
+}
+
+int map_init(struct map *m, size_t size) {
+	m->table = calloc(size, sizeof(struct entry));
+	m->size = size;
+
+	/** Ensure map was allocated. */
+	if (m->table == NULL) return -1;
+	else return 0;
+}
+
+void map_free(struct map *m) {
+	free(m->table);
+}
+
+int map_put(struct map *m, void *key, void *value, size_t size) {
+	size_t idx = fnv_1a(key, size) % m->size;
+	size_t start = idx;
+
+	
+	while (m->table[idx].key != NULL) {
+		idx = (++idx) % m->size;
+		if (idx == start) return -1;
```

Comment:
If capacity is reached, then we don't want to just reject the request. You will need to create a `resize` function that increases the capacity when a certain load factor is hit so capacity is never reached. For our purposes, the load factor for resizing should be >= 0.5 and the capacity should double.

Reply:
oh i forgot about this lol